subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveTaskExecutionModeExecuter.java

```diff
@@ -33,24 +37,32 @@
 public class ResolveTaskExecutionModeExecuter implements TaskExecuter {
     private static final Logger LOGGER = LoggerFactory.getLogger(ResolveTaskExecutionModeExecuter.class);
 
+    private final PropertyWalker propertyWalker;
     private final TaskExecuter executer;
     private final TaskExecutionModeResolver executionModeResolver;
+    private final FileCollectionFactory fileCollectionFactory;
 
-    public ResolveTaskExecutionModeExecuter(TaskExecutionModeResolver executionModeResolver, TaskExecuter executer) {
+    public ResolveTaskExecutionModeExecuter(TaskExecutionModeResolver executionModeResolver, FileCollectionFactory fileCollectionFactory, PropertyWalker propertyWalker, TaskExecuter executer) {
+        this.fileCollectionFactory = fileCollectionFactory;
+        this.propertyWalker = propertyWalker;
         this.executer = executer;
         this.executionModeResolver = executionModeResolver;
     }
 
     @Override
     public TaskExecuterResult execute(final TaskInternal task, TaskStateInternal state, final TaskExecutionContext context) {
         Timer clock = Time.startTimer();
-        TaskExecutionMode taskExecutionMode = executionModeResolver.getExecutionMode(task, context.getTaskProperties());
+        TaskProperties properties = DefaultTaskProperties.resolve(propertyWalker, fileCollectionFactory, task);
+        context.setTaskProperties(properties);
+        TaskExecutionMode taskExecutionMode = executionModeResolver.getExecutionMode(task, properties);
+
         context.setTaskExecutionMode(taskExecutionMode);
         LOGGER.debug("Putting task artifact state for {} into context took {}.", task, clock.getElapsed());
         try {
             return executer.execute(task, state, context);
         } finally {
             context.setTaskExecutionMode(null);
+            context.setTaskProperties(null);
```

Comment:
I think this is not necessary.

Reply:
I reverted the commit. I don't plan on polishing the revert. I'll revert the revert on `master`.